Loop/Managers/DeviceDataManager.swift

```diff
@@ -1424,8 +1424,39 @@ extension DeviceDataManager {
                 }
             }
         }
-        // Wait up to 25 seconds for uploads triggered by these commands to finish
-        let _ = remoteDataServicesManager.waitForUploadsToFinish(timeout: .now() + TimeInterval(25))
+    }
+    
+    func activateRemoteOverride(_ remoteOverride: TemporaryScheduleOverride?) {
+        loopManager.mutateSettings { settings in settings.scheduleOverride = remoteOverride }
+        self.triggerBackgroundUpload(for: .overrides)
+    }
+    
```

Comment:
For remote bolus and remote carbs -- there is an additional background task we do. We need to wait for the carb entry or bolus entry to complete first before we wait for the upload action. Without this, the app was sometimes hanging earlier in the process.